### PR TITLE
ci: add script for private roachtest nightly

### PR DESCRIPTION
**Pull request by rickystewart**
Part of: DEVINFHD-809.
Epic: none
Release note: None

### CODE REVIEW

**Comment by rickystewart** [comment]
[TeamCity job](https://teamcity.cockroachdb.com/buildConfiguration/Internal_Cockroach_Development_PrivateRoachtest?mode=builds).

**Review by rail — APPROVED** [review]

**Comment by rickystewart** [comment]
bors r=rail

File: build/teamcity/internal/cockroach/nightlies/private_roachtest.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+set -exuo pipefail
+
+dir="$(dirname $(dirname $(dirname $(dirname $(dirname "${0}")))))"
+
+source "$dir/teamcity-support.sh"  # For $root
+source "$dir/teamcity-bazel-support.sh"  # For run_bazel
+
+BAZEL_SUPPORT_EXTRA_DOCKER_ARGS="-e LITERAL_ARTIFACTS_DIR=$root/artifacts -e BUILD_VCS_NUMBER -e CLOUD=gce -e TESTS -e COUNT -e GITHUB_API_TOKEN -e GITHUB_ORG -e GITHUB_REPO -e GOOGLE_EPHEMERAL_CREDENTIALS -e SLACK_TOKEN -e TC_BUILDTYPE_ID -e TC_BUILD_BRANCH -e TC_BUILD_ID -e TC_SERVER_URL" \
```
Comment (jlinder) [inline build/teamcity/internal/cockroach/nightlies/private_roachtest.sh:10]: Why pass `TESTS` if it's not used? Or, why set `costfuzz/workload-replay` in the impl and not set it as the value for `TESTS`?

Comment (rickystewart) [reply build/teamcity/internal/cockroach/nightlies/private_roachtest.sh:10]: You're right, this is odd. Makes no difference right now but I will fix this in a follow-up.

**Pull request merged**